redis-gin-mongo/src/web_api/main.go

```
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/gin-gonic/gin"
+	"github.com/go-redis/redis/v8"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+type BlogPost struct {
+	Title  string `json:"title"`
+	Author string `json:"author"`
+	Body   string `json:"body"`
+}
+
+type Doc struct {
+	Data bson.D `json:"data"`
+}
+
+type Docs struct {
+	Data []bson.M `json:"data"`
+}
+
+var REDIS_HOST = os.Getenv("REDIS_HOST")
+var REDIS_PORT = os.Getenv("REDIS_PORT")
+var ANALYTICS_SERVICE_HOST = os.Getenv("ANALYTICS_SERVICE_HOST")
+var ANALYTICS_SERVICE_PORT = os.Getenv("ANALYTICS_SERVICE_PORT")
+var BLOG_SERVICE_HOST = os.Getenv("BLOG_SERVICE_HOST")
+var BLOG_SERVICE_PORT = os.Getenv("BLOG_SERVICE_PORT")
+
+func getDoc(c *gin.Context) {
+	title := c.Query("title")
+	address := fmt.Sprintf("http://%s:%s/get-doc?title=%s", BLOG_SERVICE_HOST, BLOG_SERVICE_PORT, title)
+	resp, _ := http.Get(address)
```

err is being ignored here (possible 503 for example / 404)